docs/howto/manage-charm-design.md

```diff
@@ -0,0 +1,65 @@
+(how-to-manage-charm-design)=
+# How to manage charm design
```

Comment:
This title is very artificial. Also, I think it shouldn't exist and its two sections should be moved into the Manage charms doc. That was precisely the idea behind that doc -- to help tell this high level how-to story about how to lay out the pieces in your charm code and how to handle errors, etc. I think the manage logs doc can probably be incorporated as a section into the Manage charms doc too. So:

Manage charms
- Design/Architect a charm
- Handle logs in a charm (dunno if "handle" is the best word there)
- Handle errors in a charm
- Build functionality in a charm (this can be the title for the section that says "a charm should do one thing and do it well", and "a charm should take advantage of other charms")